In [26]:
import numpy as np
from itertools import permutations
from tqdm.auto import trange, tqdm

In [27]:
2*3*4*5*6*7*8*9*10

3628800

In [74]:
eps = 0.000001
for i in tqdm(np.arange(345, 347)):
    e = i*i*i
    perm = permutations(list(str(e)))
    l = set()
    for p_str in perm:
        p = int(''.join(p_str))
        p_crt = float(p)**(1/3)
        if(np.abs(p_crt - np.round(p_crt))) < eps:
            l.add(int(np.round(p_crt)))
    if len(l)>=3:
        print('-'*10)
        print(i, e)
        print(l)
        print([k**3 for k in l])

  0%|          | 0/2 [00:00<?, ?it/s]

----------
345 41063625
{384, 345, 405}
[56623104, 41063625, 66430125]


The cube 41063625 , can be permuted to produce two other cubes: 56623104 and 66430125. In fact, 41063625 is the smallest 
cube which has exactly three permutations of its digits which are also cube. 

Could you produce a python script that finds the smallest cube for which exactly five permutations of its digits are cube ?

In [76]:
from collections import defaultdict

def get_cube_permutations(n):
    cube_dict = defaultdict(list)
    cube_count = defaultdict(int)

    i = 1
    while True:
        cube = i ** 3
        cube_str = ''.join(sorted(str(cube)))
        cube_dict[cube_str].append(cube)
        cube_count[cube_str] += 1

        if cube_count[cube_str] == n:
            return min(cube_dict[cube_str])

        i += 1

result = get_cube_permutations(6)
print(f"The smallest cube with exactly five cube permutations is: {result}")

The smallest cube with exactly five cube permutations is: 1426487591593
